src/lib/assets/Heatmap.ts

```diff
@@ -0,0 +1,165 @@
+import {
+  CanvasTexture,
+  PlaneGeometry,
+  MeshBasicMaterial,
+  Mesh,
+  Color,
+  Scene,
+  Vector3
+} from 'three';
+import { ReferencePlane } from './ReferencePlane';
+
+export class Heatmap {
+  private canvas: HTMLCanvasElement;
+  private ctx: CanvasRenderingContext2D | null;
+  private texture: CanvasTexture;
+  public scene: Scene;
+  private referencePlane: ReferencePlane;
+  private planeGeometry: PlaneGeometry;
+  private planeMaterial: MeshBasicMaterial;
+  public plane: Mesh;
+  private geometryMode: string;
+
+  public minValue: number;
+  public maxValue: number;
+  public minHue: number;
+  public maxHue: number;
+
+  constructor(scene: Scene, referencePlane: ReferencePlane) {
+    this.scene = scene;
+    this.canvas = document.createElement('canvas');
+    this.ctx = this.canvas.getContext('2d');
+    this.texture = new CanvasTexture(this.canvas);
+    this.referencePlane = referencePlane;
+    this.planeGeometry = new PlaneGeometry(referencePlane.size, referencePlane.size);
+    this.planeGeometry.rotateX( - Math.PI / 2 );
+    this.planeGeometry.translate(0,0.01,0)
+    this.planeMaterial = new MeshBasicMaterial({ map: this.texture, transparent: true, opacity: 0.9, alphaHash: true });
+    this.plane = new Mesh(this.planeGeometry, this.planeMaterial);
+    this.plane.visible = false; // Hidden by default
+    this.scene.add(this.plane);
+
+    // Resolution of the heatmap
+    this.canvas.width = this.referencePlane.size;
+    this.canvas.height = this.referencePlane.size;
+  }
+
+  updateHeatmap(values: Array<[Vector3, number]>) {
```

Comment:
This function is not used, because `updateHeatmapAdvanced` is used instead, but I'll still leave it just for the record.